firmware/cartridges/zaxxon.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2020 Christian Graefe
+ * Copyright (c) 2019-2020 Kim Jørgensen
+ *
+ * This software is provided 'as-is', without any express or implied
+ * warranty.  In no event will the authors be held liable for any damages
+ * arising from the use of this software.
+ *
+ * Permission is granted to anyone to use this software for any purpose,
+ * including commercial applications, and to alter it and redistribute it
+ * freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not
+ *    claim that you wrote the original software. If you use this software
+ *    in a product, an acknowledgment in the product documentation would be
+ *    appreciated but is not required.
+ * 2. Altered source versions must be plainly marked as such, and must not be
+ *    misrepresented as being the original software.
+ * 3. This notice may not be removed or altered from any source distribution.
+ */
+
+/*************************************************
+* C64 bus read callback
+*************************************************/
+static inline bool zaxxon_read_handler(uint8_t control, uint16_t addr)
+{
+    if (!(control & C64_ROML))
+    {
+        // switch upper bank
+        crt_ptr = crt_banks[addr & 0x1000 ? 1 : 0];
```

Comment:
This cartridge is quite simple, but in general you want the data to be written as fast as possible on the bus, so I would move this line below the c64_data_write.

Reply:
Yes, good point! The bank switching should be moved down.